### PR TITLE
chore: Deprecate DND feature flags

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
This PR:
- removes `ENABLE_DND_WITH_CLICK_UX` feature flag and consolidates its behaviour with `ENABLE_EXPLORE_DRAG_AND_DROP` ff
- deprecates `ENABLE_EXPLORE_DRAG_AND_DROP` (true by default - to be removed in 4.0)

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/orgs/apache/projects/201/views/1?pane=issue&itemId=17888760
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/dndControls.tsx [hunk 1]
```diff
@@ -244,8 +244,7 @@ export const dndGranularitySqlaControl: typeof dndSeriesControl = {
   default: (c: Control) => c.default,
   clearable: false,
   canDelete: false,
-  ghostButtonText: t('Drop temporal column here'),
-  clickEnabledGhostButtonText: t('Drop a temporal column here or click'),
+  ghostButtonText: t('Drop temporal column here or click'),
```
Comment (villebro) [inline superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/dndControls.tsx:247]: Was leaving the "a" out intentional? I'm ok both ways but if we change the string existing translations will break.

**Review by villebro — APPROVED** [review]
LGTM with a minor wording/translation comment

File: superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/dndControls.tsx (hunk 1 above)
Comment (kgabryje) [reply superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/dndControls.tsx:247]: Good eye! Definitely not intentional

**Pull request merged**